### PR TITLE
Admin homepage posting card

### LINKED ISSUE #343: Admin homepage super admin posting card
### Summary and Background
Posting cards are used through the admin homepage. For Super Admins, they lead to various flows and change based on posting state (draft, unscheduled, scheduled, past).

### User Story
As a Super Admin, I want to be able to view and interact with my postings.

### Solution (Core Elements)

This ticket does not cover accessing other flows, only the posting card UI and the make a copy functionality.

Each posting card outlines the posting branch, title, date, number of registrants, deadline, and access to different flows. This is dependent on the posting state (draft, unscheduled, scheduled, past).

1. Drafts are postings that the super admin began to create but did not finish. Super admin can edit the draft, make a copy, or delete the posting. If a posting detail hasn't been filled yet, put placeholder text (i.e Untitled posting).

2. Unscheduled postings have been posted but no schedule has been made for them yet by the super admin. They can be live or past deadline. Super admin can edit or make a copy of the posting.

3. Scheduled postings have their schedules reviewed and published but is not past the event date. Super admin can make a copy or delete the posting.

4. Past postings have completed their shifts - past end date. Super admin can make a copy or delete the posting.

Copied postings populate in the Drafts tab.

### Mockups

Drafts:

Unscheduled:

Scheduled:

Past:

Other designs: 
https://www.figma.com/file/TWJTNWjK75A6VgSisySTLe/Sistering-%2F-Admin-Experience?node-id=1420%3A39607

More info on posting states:
https://www.figma.com/file/TWJTNWjK75A6VgSisySTLe/Sistering-%2F-Admin-Experience?node-id=937%3A16935

### Technical

Create new component that takes the appropriate props to render card under `frontend/components/admin/SuperAdminPostingCard.tsx` as per design

### LINKED ISSUE #344: Admin homepage admin posting card
### Summary and Background
Posting cards are used through the admin homepage. For admins, they lead to various flows and change based on posting state (scheduled, past).

### User Story
As an Admin, I want to be able to view and interact with my postings.

### Solution (Core Elements)

This ticket does not cover accessing other flows, only the posting card UI. A lot of this ticket can be reused from the super admin posting card (#343).

Each posting card outlines the posting branch, title, date, number of registrants, deadline, and access to different flows. This is dependent on the posting state (scheduled, past).

1. Scheduled postings have their schedules reviewed and published but is not past the event date.

2. Past postings have completed their shifts - past end date.

### Mockups

Scheduled:

Past:

Other designs: 
https://www.figma.com/file/TWJTNWjK75A6VgSisySTLe/Sistering-%2F-Admin-Experience?node-id=1420%3A39607

More info on posting states:
https://www.figma.com/file/TWJTNWjK75A6VgSisySTLe/Sistering-%2F-Admin-Experience?node-id=937%3A16935

### Technical

Create new component that takes the appropriate props to render card under `frontend/components/admin/AdminPostingCard.tsx` as per design

### PR DESCRIPTION
**Pull request by lambo-liu**
## Ticket link

Closes #343 
Closes #344 

## Implementation description
* Implemented the Admin / Super Admin posting cards
* Rendering depends on the Posting's states, see [Figma](https://www.figma.com/file/TWJTNWjK75A6VgSisySTLe/Sistering-%2F-Admin-Experience?node-id=1420%3A39607)
* Backend posting status doesn't correspond with frontend, so they are resolved as discussed w/ Rickson. See getPostingStatus() method for details.
* Preview:
Screen Shot 2022-06-25 at 7 29 18 PM

## Steps to test
1. Navigate to http://localhost:3000/ with an admin account (this should also work for employees, but backend currently unauthorizes employees to retrieve postings).
2. Check that the cards render according to designs. Can try adding more postings with different statuses.

## What should reviewers focus on?
* Correctness of rendering / matches designs

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

**Comment by lambo-liu** [comment]
For terminology in this ticket, we assume that (admin --> super admin) and (employee --> admin).

**Review by MatoPlus — COMMENTED** [review]
Looks good at the surface. Will dig deeper into review today. Do let me know when this gets merged so I can do the migration on staging.

File: frontend/src/components/pages/Default.tsx [hunk 1]
```diff
@@ -74,25 +93,30 @@ const Default = (): React.ReactElement => {
...
+                autoClosingDate={posting.autoClosingDate}
+                branchName={posting.branch.name}
+                numVolunteers={posting.numVolunteers}
+                navigateToAdminSchedule={() =>
+                  navigateToAdminSchedule(posting.id)
+                }
+              />
+            </Box>
+          ))}
+      </SimpleGrid>
```
Comment (MatoPlus) [inline frontend/src/components/pages/Default.tsx:118]: @briantu relevant for your homepage query/integration ticket

File: frontend/src/components/admin/AdminPostingCard.tsx [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+              formatDateStringYear(autoClosingDate)
+            )}
+          </Text>
+          {status === PostingStatus.DRAFT ||
+          status === PostingStatus.UNSCHEDULED ? (
+            <Button
+              variant="ghost"
+              disabled={
+                status === PostingStatus.DRAFT
+              } /* TODO: Link to review registrants page */
```
Comment (MatoPlus) [inline frontend/src/components/admin/AdminPostingCard.tsx:128]: Is this basically to the `admin/schedule/posting/{id}` route?

File: frontend/src/components/admin/AdminPostingCard.tsx [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+              <MenuButton
+                as={IconButton}
+                aria-label="Options"
+                icon={<Icon as={BsThreeDots} w={6} />}
+                variant="unstyled"
+              />
+              <MenuList shadow="md">
+                <MenuItem>Edit</MenuItem>
+                <MenuItem>Make a copy</MenuItem>
+                <MenuItem>Delete</MenuItem>
```
Comment (MatoPlus) [inline frontend/src/components/admin/AdminPostingCard.tsx:77]: Can you actually attach the `DeleteModal` to open on the click handler? It should just be adding a state and passing in the args you want for the text you want to show in the modal

**Review by MatoPlus — CHANGES_REQUESTED** [review]
small request and LGTM!

File: frontend/src/components/volunteer/PostingCard.tsx [hunk 4]
```diff
@@ -82,23 +79,19 @@ const PostingCard = ({
...
         <HStack spacing={4}>
           <Text textStyle="caption" noOfLines={2}>
-            <TimeIcon w={4} pr={1} />
-            {type === "OPPORTUNITY"
-              ? "See posting details"
-              : `${formatTimeHourMinutes(
-                  new Date(eventFirstTime),
-                )} - ${formatTimeHourMinutes(new Date(eventLatestTime))}`}
+            <TimeIcon w={6} pr={2} />
+            See Posting Details
```
Comment (MatoPlus) [inline frontend/src/components/volunteer/PostingCard.tsx:88]: can you revert this?

File: frontend/src/components/volunteer/PostingCard.tsx [hunk 5]
```diff
@@ -82,23 +79,19 @@ const PostingCard = ({
...
-            <CalendarIcon w={4} pr={1} />
-            {type === "OPPORTUNITY"
-              ? `${formatDateStringYear(startDate)} - ${formatDateStringYear(
-                  endDate,
-                )}`
-              : formatDateStringYear(startDate)}
+            <CalendarIcon w={6} pr={2} />
+            {`${formatDateStringYear(startDate)} - ${formatDateStringYear(
+              endDate,
+            )}`}
```
Comment (MatoPlus) [inline frontend/src/components/volunteer/PostingCard.tsx:94]: can you revert this as well?

**Review by MatoPlus — CHANGES_REQUESTED** [review]
LGTM aside from the comments. I think you're the implementation we want : (

**Review by MatoPlus — APPROVED** [review]
Ship it ⚓🛳️

**Pull request merged**